File: packages/amplify_datastore/ios/Classes/types/hub/FlutterNetworkStatusEvent.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation
import Amplify
import AmplifyPlugins

struct FlutterNetworkStatusEvent: FlutterHubEvent {
    var eventName: String
    var active: Bool

    init(payload: HubPayload) throws {
        guard let networkStatus = payload.data as? NetworkStatusEvent else {
                  throw FlutterDataStoreError.hubEventCast
              }
        self.eventName = shortEventName(eventName: payload.eventName)
        self.active = networkStatus.active
    }

    func toValueMap() -> [String: Any] {
        return [
            "eventName": eventName,
            "active": active
        ]
    }
}

File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterNetworkStatusEvent.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.hub

class FlutterNetworkStatusEvent(override val eventName: String, private val active: Boolean) : FlutterHubEvent {
    override fun toValueMap(): Map<String, Any> {
        return mapOf(
            "eventName" to this.eventName,
            "active" to this.active
        )
    }
}
